Corona-Warn-App/src/main/java/de/rki/coronawarnapp/appconfig/ConfigChangeDetector.kt

```diff
@@ -60,4 +64,19 @@ class ConfigChangeDetector @Inject constructor(
             RiskLevelRepository.setRiskLevelScore(RiskLevel.UNDETERMINED)
         }
     }
+
+    @VisibleForTesting
+    internal suspend fun checkForDiagnosisKeysDataMapping(exposureWindowRiskCalculationConfig: ExposureWindowRiskCalculationConfig) {
+        val oldDiagnosisKeyDataMapping = enfClient.getDiagnosisKeysDataMapping()
+        val newDiagnosisKeyDataMapping = exposureWindowRiskCalculationConfig.diagnosisKeyDataMapping
+
+        if (oldDiagnosisKeyDataMapping != newDiagnosisKeyDataMapping) {
```

Comment:
Does this compare object references or compare by value?

Reply:
should be an equal comparison, not 100% sure if its enough though, will do some checking

Reply:
Comparison has been changed and is now unittested